docs/quality-assurance-and-test.md

```diff
@@ -4,43 +4,147 @@ Welcome to the introduction on how we are working with Quality Assurance and Tes
+
+### Test activities and types
+
+There are a lot of frameworks and definitions of testing activities, and here we will outline the concepts used within Green Energy Hub. Some of these activities are only done on indivdual forks - but please see this as an inspiration to how you as an individual contributor or organisation can organize your quality activities. Depending on your contribution or task the coverage on different test activities might differ, and some might not be applicable. We do not want you to test for the sake of testing - but share the experiences and results of testing, so that the testing is a collaborative activity, where the community are open to share concerns, ideas and new ways of thinking.
+
+Are you using TTD or BDD or other types of code practises please be inspirational - but respect the difference in ways of working as well.
+
+(NB! The list is not a sequential representation of test activities)
+
+- Peer Review (Mandatory activity in Green Energy Hub PR gate)
+
+- Unit Test (Mandatory activity in Green Energy Hub PR gate)
+
+- Component Test
+
+- Infrastructure Test
+
+- Internal Domain Integration Test
+
+- Domain System Test
+
+- Contract Test
+
+- Inter-domain Integration Test
+
+- Performance Test
+
+- Security Test
+
+- Automated Regression Test
+
+- Business Process Test
 
 ## Tools
 
 In Green Energy Hub we need to share and delevop our ways of working - here will be of list of tools being used.
 
 ### Code Coverage
 
-Unit test code coverage tool.
+We are currently using Coverlet in our PR gate to ensure a 80% unit test code coverage. This is not blocking for PR's - yet - but we would like all to consider how to gain quality where possible, and also be true to the expectation of challenging your own work and others. If you don't reach the target, consider why, what could increase your coverage and have you used different coverage types when designing your unit test.
+
+#### Types of Code Coverage
 
-### CI/CD Inspiration
+- Line Coverage
+100% line coverage are exercised when covering at least one element of each line.
 
-How to do CI/CD.
+- Statement Coverage
+100% statement coverage is achieved when covering all statements in the code.
 
-## Test Data
+- Decision Coverage
+100% Decision Coverage are exercised when all possible outcomes of decision statements are covered
 
-Green Energy Hub would like to publish test data for use and inspiration.
+- Path Coverage
+Same approach as decision coverage, whereas decision coverage only test for all possible outcomes are functional - path coverage are to determine how the business logic are functional across multiple decisions. Path coverage will automatically require decisions being tested multiple times in different paths, and the test coverage for decision coverage will exceed 100% with path coverage.
+
```

Comment:
Should we include method coverage or are we not using this measurement?

Reply:
I don't have a dinition on method coverage - do you? Are you usind this method within the project - please add it.

Reply:
Method coverage is just how many of all methods have been called.

Reply:
Coverlet uses line, branch and method coverage as it's default evaluation methods.

Reply:
Like this:
" - Method Coverage
100% Method coverage is achieved when all methods have been called"

What about Branch coverage - should we define that?

Reply:
Adapted to match code cov's coverage types